sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/socket.scala

```diff
@@ -136,7 +149,8 @@ class TextSocketSourceProvider extends StreamSourceProvider with DataSourceRegis
       parameters: Map[String, String]): Source = {
     val host = parameters("host")
     val port = parameters("port").toInt
-    new TextSocketSource(host, port, sqlContext)
+    new TextSocketSource(host, port,
+      parameters.getOrElse("includeTimestamp", "false").toBoolean, sqlContext)
```

Comment:
sourceSchema and createSource are separate functions that are passed the parameters map, so the conversion to Boolean needs to happen in both places. Think it's fine if this throws an error if it's not parseable ... the line above will also throw an error if "port" is not parseable as Int.